src/cpp/src/device_config.hpp

```diff
@@ -29,6 +29,20 @@ class DeviceConfig {
         if (m_device == "CPU") {
             auto inference_precision = core.get_property(device, ov::hint::inference_precision);
             m_kv_cache_type = inference_precision == ov::element::bf16 ? ov::element::bf16 : ov::element::f16;
+            // if user sets precision hint, kv cache type should be changed
+            if (plugin_config.find(ov::hint::inference_precision.name()) != plugin_config.end()) {
+                const auto& type_name = plugin_config.at(ov::hint::inference_precision.name()).as<std::string>();
```

Comment:
```suggestion
                const auto& type_name = plugin_config.at(ov::hint::inference_precision.name()).as<ov::element::Type>();
```
w/o any string conversions

Reply:
Done.